### PR TITLE
feat: Read docx tables

### PR DESCRIPTION
**Pull request by KevinRPan**
First commit. Reviewed Contributing.md but please let me know if I've missed anything. 

For CHANGELOG.md, I think my editor may have automatically changed bullet formatting. Also not sure if this counts as a core code change. 

My only line change was under Features: 
- Added table reading as html with URL parsing to `partition_docx` in docx

Can test with partition_docx(fake_table.docx)

Make test seems to have some random pandoc errors on my machine but not related for the file changed.

### CODE REVIEW

**Comment by KevinRPan** [comment]
@qued @MthwRobinson

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,30 +2,31 @@
 
 ### Enhancements
 
-* Added an additional trace logger for NLP debugging.
+- Added an additional trace logger for NLP debugging.
```
Comment (cragwolfe) [inline CHANGELOG.md:5]: doc nit, looks like the .md changes were auto changed by an editor? please revert.

File: unstructured/partition/docx.py [hunk 2]
```diff
@@ -92,15 +127,44 @@ def partition_docx(
...
-            element.metadata = ElementMetadata(filename=metadata_filename)
-            elements.append(element)
-
+    table_index = 0 
+    
+    for element_item in document.element.body:
+        if element_item.tag.endswith('tbl'):
+            table = document.tables[table_index]
+            html_table = _convert_table_to_html(table)
+            element = Text(html_table)
```
[the remark this answers is not in the document]
Comment (MthwRobinson) [reply unstructured/partition/docx.py:136]: Let's also make the text of the elements the raw text for the table, and we'll put the HTML structured for the table in `text_as_html`. That will make it consistent with the behavior for PDFs.

If we're going to represent all tables as HTML tables, I actually may like having the HTML as the `text` attribute better than the plain text. But I'd rather make that change in a separate PR if we're going to make it.

For PDFs, run the following code

```python
from unstructured.partition.pdf import partition_pdf

elements = partition_pdf("example-docs/layout-parser-paper.pdf", infer_table_structure=True)
tables = [el for el in elements if el.category == "Table"]
print(tables[0].text)
print(tables[0].metadata.text_as_html)
```

and you'll get for `text`.

```
Dataset Base Model1 Large Model Notes PubLayNet [38] F / M M Layouts of modern scientiﬁc documents PRImA [3] M - Layouts of scanned modern magazines and scientiﬁc reports Newspaper [17] F - Layouts of scanned US newspapers from the 20th century TableBank [18] F F Table region on modern scientiﬁc and business document HJDataset [31] F / M - Layouts of history Japanese documents
```

and this for `metadata.text_as_html`

```
<table><thead><th>Dataset</th><th>| Base Model’</th><th>Large Model</th><th>| Notes</th></thead><tr><td>PubLayNet</td><td>[38] F/M</td><td>M</td><td>Layouts of modern scientific documents</td></tr><tr><td>PRImA [3]</td><td>M</td><td></td><td>Layouts of scanned modern magazines and scientific reports</td></tr><tr><td>Newspaper</td><td>F</td><td></td><td>Layouts of scanned US newspapers from the 20th century</td></tr><tr><td>TableBank</td><td>F</td><td></td><td>Table region on modern scientific and business document</td></tr><tr><td>HJDataset</td><td>[31] F/M</td><td></td><td>Layouts of history Japanese documents</td></tr></table>
```

**Review by MthwRobinson — CHANGES_REQUESTED** [review]
@KevinRPan - Great PR, thanks a lot for submitting this! Have a few small inline comments in addition to @cragwolfe 's comment about the changelog. Also, could you add a test to `test_unstructured/partition/test_docx.py`?

**Comment by MthwRobinson** [comment]
You can run `make tidy` to fix those linting errors in CI.

File: CHANGELOG.md (hunk 1 above)
Comment (KevinRPan) [reply CHANGELOG.md]: Fixed, thanks

File: unstructured/partition/docx.py [hunk 3]
```diff
@@ -60,6 +65,36 @@
     "Title": Title,
 }
 
+# NOTE(kevinrpan) there are table styles in the docx library that I am not applying here
+
+# NOTE(kevinrpan) Open to other ways of doing this 
+def GetParagraphRuns(paragraph):
```
[the remark this answers is not in the document]
Comment (KevinRPan) [reply unstructured/partition/docx.py]: yup updated

File: unstructured/partition/docx.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (KevinRPan) [reply unstructured/partition/docx.py]: Good call, I missed that there was a table element. 

Thanks for the example. I made an attempt at this update to add the text of the table as a Table element without HTML, with the html table as a metadata element. 

Please let me know if I've understood this correctly.

File: unstructured/partition/docx.py [hunk 4]
```diff
@@ -92,15 +127,44 @@ def partition_docx(
...
+    table_index = 0 
+    
+    for element_item in document.element.body:
+        if element_item.tag.endswith('tbl'):
+            table = document.tables[table_index]
+            html_table = _convert_table_to_html(table)
+            element = Text(html_table)
+            if element is not None:
+                element.metadata = ElementMetadata(
+                    filename=metadata_filename)
```
[the remark this answers is not in the document]
Comment (KevinRPan) [reply unstructured/partition/docx.py]: updated

File: unstructured/partition/docx.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MthwRobinson) [reply unstructured/partition/docx.py]: Updated behavior looks great, thanks!

**Review by MthwRobinson — APPROVED** [review]
Verified the updates, LGTM! Will merge once CI passes.

**Pull request merged**